stl/inc/vector

```diff
@@ -434,6 +434,32 @@ constexpr _Ty* _Unfancy_maybe_null(_Ty* _Ptr) noexcept { // do nothing for plain
     return _Ptr;
 }
 
+extern "C" {
+void __sanitizer_annotate_contiguous_container(
+    const void* _First, const void* _End, const void* _Old_last, const void* _New_last) noexcept;
+extern const bool _Asan_vector_should_annotate;
+}
+
+#pragma comment(linker, \
+    "/alternatename:__sanitizer_annotate_contiguous_container=__sanitizer_annotate_contiguous_container_default")
+#pragma comment(linker, "/alternatename:_Asan_vector_should_annotate=_Asan_vector_should_annotate_default")
+
+#ifdef __SANITIZE_ADDRESS__
+#pragma comment(lib, "stl_asan")
+#endif // __SANITIZE_ADDRESS__
+
+#if defined(__SANITIZE_ADDRESS__) || defined(_ANNOTATE_VECTOR)
+template <class _Vec, class = void>
+_INLINE_VAR constexpr bool _Has_minimum_allocation_alignment =
+    disjunction_v<bool_constant<alignof(typename _Vec::value_type) >= 8>,
+        is_same<typename _Vec::allocator_type, allocator<typename _Vec::value_type>>>;

```

Comment:
Is the minimum allocation of `std::allocator` actually 8 bytes when not linked against the ASan runtime? If it is then it's fine. Otherwise I think `_Has_minimum_allocation_alignment` should remain as is.

Reply:
Presumably we would conditionally define `_Minimum_allocation_alignment` within `std::allocator` only when `__SANITIZE_ADDRESS__`. It seems weird to me to have an opt-in mechanism for allocators, but not use it for `std::allocator`.

Reply:
Sounds good to me.